hierarchicalforecast/methods.py

```diff
@@ -966,6 +970,35 @@ def fit_predict(self,
 
     __call__ = fit_predict
 
+@njit(parallel=True, fastmath=True, error_model="numpy")
```

Comment:
```suggestion
@njit("f8[:, :](f8[:, :], b1[:, :])", nogil=True, cache=True, parallel=True, error_model="numpy", fastmath=True)
```

Do you think it would be worth it to use eager compilation by specifying the signature and caching? I guess in the vast majority of cases the function would only be called once to try out the `mint_cov` method, so most likely not.

Anyway, you may also want to experiment with releasing the GIL to see if it makes a difference when benchmarking as the function is parallelised.

Reply:
See earliier comment with benchmarks:
- Static typing generally slows it down, even when accounting for contiguity of arrays
- `nogil=True` makes sense
- `cache=True` makes sense

Thanks!